promtimer/promtimer.py

```diff
@@ -375,6 +379,22 @@ def main():
         times = cbstats.CBCollect.compute_min_and_max_times(stats_sources)
         min_time = datetime.datetime.fromtimestamp(times[0]).isoformat()
         max_time = datetime.datetime.fromtimestamp(times[1]).isoformat()
+
+        timezone_override = os.environ.get('GRAFANA_TZ', '')
+        if timezone_override != '':
+            default_timezone = timezone_override
+        else:
+            try:
+                default_timezone = next(
+                    map(lambda s: s.get_timezone(), stats_sources), None)
```

Comment:
It bothers me a bit that if we have 20 cbcollects we'll open 20 `couchbase.log`s and parse them. If you'd like to keep the functional style, you could use `functools.reduce()`. Something like:
```
    functools.reduce(lambda: tz, s: tz if tz is not None else s.get_timezone(), None)
```

Reply:
Done